### DISCUSSION TITLE
Allow separate resizing of array observables that should have the same length

### DISCUSSION THREAD

**Discussion by jkrumbiegel** [JuliaPlots/Makie.jl#4169] [Ideas]
Just writing this down for vizcon again. It's complicated to resize plots like `Scatter` if you can't change the number of points without them going out of sync with the number of colors. Currently, doing so just breaks the plot or results in undefined behavior. It would be better to allow separate resizing and to simply disable a plot from drawing until its different buffers have been synced again. Then it would be possible to do this:

```julia
coords = Node(rand(10, 2))
colors = Node(rand(RGBf0, 10))

scatter(coords, color = colors)

coords[] = rand(20, 2) # this will fail now or cause weirdness
# ideally at this point the plot should just not draw, be in a "suspended state"
colors[] = rand(RGBf0, 20)
# and now it should be fine again
```

**Comment by asinghvi17** [comment]
I was thinking about this yesterday, and while the solution you propose makes more sense, I kind of wonder if this might also work.

Consider an Array of Observables, in which you mutate each using `o.val = ...`, and then update the first one, which will trigger a lift within the plot function.  Would this do the trick?  We could also make a convenience struct for this.
